pkg/kubelet/kubelet.go

```diff
@@ -2176,6 +2209,10 @@ func (kl *Kubelet) syncLoopIteration(configCh <-chan kubetypes.PodUpdate, handle
 			// TODO: Do we want to support this?
 			glog.Errorf("Kubelet does not support snapshot update")
 		}
+
+		// Add sourcesReady after the operation on pod finishes
```

Comment:
I opened a separate PR #30417 for this with a more detailed explanation, since it is a more general bug fix and should probably be decoupled from this PR.